### ISSUE TITLE
Invoices - Invoice Lifecycle Tracker

### ISSUE THREAD

**Issue by PeterWong1** [bitpopulous/admin_reviews#9]
Suggest to include a "invoice lifecycle tracker" feature so the various stages of the invoice status can be followed and recorded.

![10a admin invoice lifecycle tracker](https://user-images.githubusercontent.com/31194353/37548227-5b8fc9cc-29b1-11e8-9a36-3b694a2ef71c.png)

**Comment by ArtemShatalov** [comment]
@wisdomcsharp ,

Here is my suggestion. Please check this page.

![admin-invoice-details-lifecycle](https://user-images.githubusercontent.com/12482899/37714730-f0e270aa-2d22-11e8-92ee-2c76cdbee4dc.jpg)

**Comment by PeterWong1** [comment]
@wisdomcsharp @archi-tec 

Please see my comments in the attached mark-up.

![invoice lifecycle tracker - comment](https://user-images.githubusercontent.com/31194353/37744322-4e1d0132-2da9-11e8-804a-b84ff9cb6f16.png)

It could be a useful statistic to know the breakdown in the number of invoices automatically closing as opposed to those not meeting their Goal Sale Price.  An analysis of these statistics could be carried out in the future to enhance the operations of the platform.

**Comment by ArtemShatalov** [comment]
Thank you @PeterWong1 

As for me, if we want to see the path of invoice, tree scheme shouldn't be used. That's because we can't normally show on it if one or another step was repeated multiple times. Otherwise, it becomes quite huge.

Here is an update. Line scheme almost duplicates the sequence in the table though.

![admin-invoice-details-lifecycle](https://user-images.githubusercontent.com/12482899/37876395-6c5cc33c-3054-11e8-8568-3a00dbd2e4ba.jpg)

CC'd to @wisdomcsharp

**Comment by PeterWong1** [comment]
Hi @archi-tec,

As a variation on your line option, will the enclosed suggestion work?

![lc](https://user-images.githubusercontent.com/31194353/37882752-997b42a6-30d8-11e8-87e0-64745c171599.png)

**Comment by ArtemShatalov** [comment]
Yeah @PeterWong1, your scheme is clear, thank you. But if it doesn't intend to show possible repetition, this means that shown invoice path can be incomplete as some pieces are omitted. I guess this is an important point if we're going to track the invoice path. Otherwise, this functionality probably loses its value.

Anyway, here is an update. 

1. If we want to help administrator to predict next statuses, which, I think, the purpose of tree scheme, we can add a small icon (top right corner) which opens a tree scheme for reference. I also added the next step to the line scheme.

![admin-invoice-details-lifetime-2](https://user-images.githubusercontent.com/12482899/37892099-531050fc-30df-11e8-878b-c1e479145ade.jpg)


2. I've updated the tree scheme. Probably listing the repeated iterations will help to smooth the issue.

![admin-invoice-details-lifecycle](https://user-images.githubusercontent.com/12482899/37892168-8cb3c618-30df-11e8-93e1-1c1d2869d65a.jpg)


CC'd to @wisdomcsharp

**Comment by PeterWong1** [comment]
Hi @archi-tec 

See enclosed modification of the chart for your perusal.

Listing the number of repetition as you have suggested is an idea to present the number of repetition in that loop.  The only possible repetition I can see from the invoice status flow chart is between "Auction Open" and "Auction Closed" status.  In this status, there is the option to restart the auction by the seller.  All the other locations indicates one direction only.  

@wisdomcsharp can you clarify where there will be repetition of invoice status apart from those shown in the diagram.
![lc hori](https://user-images.githubusercontent.com/31194353/37900013-978c8364-311f-11e8-939f-6845810953a3.png)

**Comment by ArtemShatalov** [comment]
@PeterWong1 , @wisdomcsharp ,

I supposed that yes, there could be repetitions between Approval - Rejected, due to mistakes: information mismatch inside the invoice copy and fields,  not signed/wrong contract uploaded.
